_episodes/08-putting-it-all-together.md

```diff
@@ -63,232 +67,264 @@ measurements. Convert any measurements in imperial units into SI units. You can
-ipython notebooks, we can make figures appear in-line within the notebook by
-writing:
+Now, let's read data and plot it!
 
 ~~~
-    %matplotlib inline
+surveys = pd.read_csv("data/surveys.csv")
+my_plot = surveys.plot("hindfoot_length", "weight", kind="scatter")
+plt.show() # not necessary in Jupyter Notebooks
 ~~~
 {: .language-python}
 
-We can start by plotting the values of a list of numbers (matplotlib can handle
-many types of numeric data, including numpy arrays and pandas DataFrames - we
-are just using a list as an example!):
+![Scatter plot of survey data set](../fig/08_scatter_surveys.png)
 
-~~~
-    list_numbers = [1.5, 4, 2.2, 5.7]
-    plt.plot(list_numbers)
-    plt.show()
-~~~
-{: .language-python}
+> ## Tip
+> By default, matplotlib creates a figure in a separate window. When using
+> Jupyter notebooks, we can make figures appear in-line within the notebook by
+> executing:
+>
+> ~~~
+> %matplotlib inline
+> ~~~
+> {: .language-python}
 
-The command `plt.show()` prompts Python to display the figure. Without it, it
-creates an object in memory but doesn't produce a visible plot. The ipython
-notebooks (if using `%matplotlib inline`) will automatically show you the figure
-even if you don't write `plt.show()`, but get in the habit of including this
-command!
+The returned object is a matplotlib object (check it yourself with `type(my_plot)`), 
+to which we may make further adjustments and refinements using other matplotlib methods.
 
-If you provide the `plot()` function with only one list of numbers, it assumes
-that it is a sequence of y-values and plots them against their index (the first
-value in the list is plotted at `x=0`, the second at `x=1`, etc). If the
-function `plot()` receives two lists, it assumes the first one is the x-values
-and the second the y-values. The line connecting the points will follow the list
-in order:
+> ## Tip
+> Matplotlib itself can be overwhelming, so a useful strategy is to 
+> do as much as you easily can in a convenience layer, _i.e._ start
+> creating the plot in Pandas or plotnine, and then use matplotlib
+> for the rest.
+{: .callout}
 
-~~~
-    plt.plot([6.8, 4.3, 3.2, 8.1], list_numbers)
-    plt.show()
-~~~
-{: .language-python}
+We will cover a few basic commands for creating and formatting plots with matplotlib in this lesson. 
+A great resource for help creating and styling your figures is the matplotlib gallery
+(<http://matplotlib.org/gallery.html>), which includes plots in many different
+styles and the source codes that create them. 
 
-A third, optional argument in `plot()` is a string of characters that indicates
-the line type and color for the plot. The default value is a continuous blue
-line. For example, we can make the line red (`'r'`), with circles at every data
-point (`'o'`), and a dot-dash pattern (`'-.'`). Look through the matplotlib
-gallery for more examples.
+
+### `plt` pyplot versus object-based matplotlib
```

Comment:
I'm not sure I understand the "comparison" (implied by "versus") here...

Reply:
we import `pyplot` module of `matplotlib` package as `plt` (`plt === matplotlib.pyplot`).
@stijnvanhoey, what were you trying to convey with this title?

...`layered plots`... have not been defined yet :)

Reply:
I actually got *versus* from the matplotlib documentation itself: https://matplotlib.org/tutorials/introductory/lifecycle.html#a-note-on-the-object-oriented-api-vs-pyplot 

Using *`plt` pyplot* to refer to what learners will typically recognise already or encounter online (e.g. stackoverflow) instead of adding more technical terms such as *state-based interface*...

Reply:
Oh, that! That's pretty "deep"!  I'd suggest moving this discussion (about stateful `plt` vs OO matplotlib) to a callout box or something (or drop entirely). Or add to instructors notes so that at least instructors are aware of such a thing.

Reply:
Maybe we should create a separate issue on how to tackle this with respect to squashing/merging?